src/backend/opencl/platform.cpp

```diff
@@ -396,43 +395,41 @@ void devprop(char* d_name, char* d_platform, char* d_toolkit, char* d_compute) {
 
     DeviceManager& devMngr = DeviceManager::getInstance();
 
-    vector<cl::Context*> contexts;
     {
         common::lock_guard_t lock(devMngr.deviceMutex);
```

Comment:
This section is only reading data from device manager, exclusive access doesn't make sense. Since we have C++14 now, we should move to shared Mutex so that multiple threads can read this info if they want to without the wait.

Alternatively, we can use boost's shared_mutex instead of std::shared_timed_mutex but we have to link against boost's thread library for that. I wonder how much of a benefit that would be - but again I would expect it to avoid timing logic if we just use `lock` and `shared_lock` methods.

Reply:
This change has been deferred to another PR after discussion.